### PR TITLE
- checking SSLOffloading header set by environment

### LINKED ISSUE #2399: For SSL Offloaded site, popups are not loading
## Description of bug
For SSL Offloaded site, popups are not loading

## Steps to reproduce
Go to any page which have pop-ups like login, register, reset-password

## Current result
Error in browser debugger. Pop up not loading

## Expected result
Popup should load as normal

## Affected version
* [x] 9.2.2
* [x] 9.2.1
* [x] 9.2
* [x] 9.1.1
* [x ] 9.1
* [x] 9.0

### PR DESCRIPTION
**Pull request by Mhtshum**
Fixes #2399 

## Summary
Checking Offloading header for popups when opening

### CODE REVIEW

File: DNN Platform/Library/Common/Utilities/UrlUtils.cs [hunk 1]
```diff
@@ -174,7 +174,8 @@ public static string[] GetQSParamsForNavigateURL()
         /// <returns>true if HTTPS or if HTTP with an SSL offload header value, false otherwise</returns>
         public static bool IsSecureConnectionOrSslOffload(HttpRequest request)
         {
-            if (request.IsSecureConnection)
+            var isRequestSSLOffloaded = IsRequestSSLOffloaded(request);
+            if (request.IsSecureConnection || isRequestSSLOffloaded)
```
Comment (Mhtshum) [inline DNN Platform/Library/Common/Utilities/UrlUtils.cs]: SSL Offloading header in load balancing  is equal to asp.net request.IsSecureConnection. Mean while this check is enough and below all code can be delete, which redundantly checking for page.IsSecure

Comment (tpluscode) [reply DNN Platform/Library/Common/Utilities/UrlUtils.cs]: I do not follow. If this was enough to check `HttpRequest#IsSecureConnection` then these change would not be necessary, right? 😉 

After all, with SSL offloading the request coming to the IIS is no longer encrypted, hence the flag would be `false`.

**Comment by Mhtshum** [comment]
Good point! let me try to explain it. Most important part is that both settings are dependent on Server configuration.  
When Webserver configure with SSL certificate and request sends as https then ASP.net Framework pick that and give HttpRequest.IsSecureConnection true, which is good for one server configuration. 

Unfortunately, when SSL offloading occurs ASP.net does'nt have any clue that this request is secure, but luckily we can set http header at load balancer end, so that when SSL offloading occurs http based request sends to backend servers with SSLOffloading header. And then any application in backend server needs to check that header to response(i.e send content) as secure request

File: DNN Platform/Library/Common/Utilities/UrlUtils.cs [hunk 2]
```diff
@@ -196,6 +197,27 @@ public static bool IsSecureConnectionOrSslOffload(HttpRequest request)
             return false;
         }
 
+        /// <summary>
+        /// check if SSL offloaded header set in application 
+        /// And also send by Load Balancer 
+        /// </summary>
+        /// <param name="request">current request</param>
+        /// <returns>true if an SSL offload header value set by application and load balancer, otherwise false</returns>
+        public static bool IsRequestSSLOffloaded(HttpRequest request)
```
[the remark this answers is not in the document]
Comment (tpluscode) [reply DNN Platform/Library/Common/Utilities/UrlUtils.cs]: Fo public use `IsSecureConnectionOrSslOffload` looks to be enough

[the remark this answers is not in the document]
Comment (tpluscode) [reply DNN Platform/Library/Common/Utilities/UrlUtils.cs]: @dnnsoftware/tag ?

**Pull request merged**